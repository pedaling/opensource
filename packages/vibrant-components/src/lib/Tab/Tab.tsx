import { Box, PressableBox } from '@vibrant-ui/core';
import { Body } from '../Body';
import { NotificationBadge } from '../NotificationBadge';
import { Title } from '../Title';
import { withTabVariation } from './TabProps';

export const Tab = withTabVariation(
  ({
    innerRef,
    title,
    borderBottomColor,
    textColor,
    update,
    description,
    id,
    onClick,
    testId = 'tab',
    active,
    ...restProps
  }) => (
    <PressableBox
      role="tab"
      as="button"
      borderWidth={0}
      ref={innerRef}
      flexDirection="row"
      borderBottomWidth={2}
      borderBottomStyle="solid"
      borderBottomColor={borderBottomColor}
      position="relative"
      cursor="pointer"
      justifyContent="center"
      alignItems="center"
      onClick={() => onClick?.(id)}
      pt={12}
      pb={10}
      backgroundColor="inherit"
      ariaSelected={active}
      data-testid={testId}
      {...restProps}
    >
      <Title level={[6, 6, 5]} weight="bold" color={textColor}>
        {title}
      </Title>
      {description ? (
        <Box ml={4}>
          <Body level={[5, 5, 3]} color={textColor} weight="medium">
            {description}
          </Body>
        </Box>
      ) : null}
      {update && (
        <Box mt={2} ml={4} display="flex" alignSelf="flex-start">
          <NotificationBadge kind="dot" size="sm" />
        </Box>
      )}
    </PressableBox>
  )
);
