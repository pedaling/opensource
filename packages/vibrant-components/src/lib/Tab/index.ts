export { Tab } from './Tab';
export type { TabProps } from './TabProps';
