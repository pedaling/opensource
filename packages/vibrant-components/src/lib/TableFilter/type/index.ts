export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
