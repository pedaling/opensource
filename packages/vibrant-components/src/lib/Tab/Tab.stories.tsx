import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { VStack } from '../VStack';
import { Tab } from './Tab';

export default {
  title: 'Tab',
  component: Tab,
  args: {
    title: 'Tab',
    active: true,
    update: true,
    id: 'tab1',
  },
} as ComponentMeta<typeof Tab>;

export const Basic: ComponentStory<typeof Tab> = props => (
  <VStack>
    <Tab {...props} />
  </VStack>
);
