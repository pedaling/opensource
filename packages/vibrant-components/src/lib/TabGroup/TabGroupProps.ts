import type { ReactElement } from 'react';
import { Box, ScrollBox, propVariant, withVariation } from '@vibrant-ui/core';
import type { TabProps } from '../Tab';

export type TabGroupProps = {
  tabId: string;
  type: 'fitContent' | 'fullWidth';
  onTabChange: (tabId: string) => void;
  children: ReactElement<TabProps>[];
};

export const withTabGroupVariation = withVariation<TabGroupProps>('TabGroup')(
  propVariant({
    props: [
      {
        name: 'type',
      },
    ],
    variants: {
      fitContent: {
        tabFlexGrow: 0,
        tabFlexShrink: 0,
        tabFlexBasis: 'auto',
        BoxComponent: ScrollBox,
        width: 'auto',
      },
      fullWidth: {
        tabFlexGrow: 1,
        tabFlexShrink: 0,
        tabFlexBasis: 'auto',
        BoxComponent: Box,
        width: '100%',
      },
    } as const,
  })
);
