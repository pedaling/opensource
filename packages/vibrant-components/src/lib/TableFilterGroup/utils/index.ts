import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
