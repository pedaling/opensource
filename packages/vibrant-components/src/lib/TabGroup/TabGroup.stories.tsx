import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Tab } from '../Tab/Tab';
import { TabGroup } from './TabGroup';

export default {
  title: 'TabGroup',
  component: TabGroup,
  args: {
    type: 'fullWidth',
  },
} as ComponentMeta<typeof TabGroup>;

export const Basic: ComponentStory<typeof TabGroup> = props => (
  <TabGroup {...props}>
    <Tab title="Tab1" id="tab1" />
    <Tab title="Tab2222" id="tab2" />
    <Tab title="Tab3" id="tab3" />
    <Tab title="Tab4" id="tab4" />
    <Tab title="Tab5" id="tab5" />
    <Tab title="Tab6" id="tab6" />
  </TabGroup>
);
