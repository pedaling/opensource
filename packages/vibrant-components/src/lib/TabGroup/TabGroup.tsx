import type { ForwardedRef, FunctionComponentElement, MouseEventHandler } from 'react';
import { Children, cloneElement, isValidElement, useCallback, useEffect, useRef, useState } from 'react';
import { useResponsiveValue } from '@vibrant-ui/core';
import { Icon } from '@vibrant-ui/icons';
import { useInView } from '@vibrant-ui/utils';
import { Divider } from '../Divider';
import { HStack } from '../HStack';
import { Paper } from '../Paper';
import { Pressable } from '../Pressable';
import { Space } from '../Space';
import { VStack } from '../VStack';
import { withTabGroupVariation } from './TabGroupProps';

const InView = ({
  onChange,
  children,
  options,
}: {
  onChange: (inView: boolean) => void;
  children: FunctionComponentElement<{ ref?: ForwardedRef<any> }>;
  options?: IntersectionObserverInit;
}) => {
  const { ref: inViewRef } = useInView({ initialInView: true, onChange, options });

  const composeRef = useCallback(
    (node: HTMLElement) => {
      if (children.ref) {
        if (typeof children.ref === 'function') {
          children.ref(node);
        } else {
          children.ref.current = node;
        }
      }

      inViewRef(node);
    },
    [children, inViewRef]
  );

  return cloneElement(children, {
    ref: composeRef,
  });
};

export const TabGroup = withTabGroupVariation(
  ({ BoxComponent, tabFlexGrow, tabFlexShrink, tabFlexBasis, tabId, onTabChange, children, ...restProps }) => {
    const tabElements = (Children.toArray(children).filter(child => isValidElement(child)) as typeof children) ?? [];
    const tabRefs = useRef<Record<string, HTMLElement>>({});
    const tabGroupRef = useRef<HTMLUListElement>(null);
    const [lastTabIsInView, setLastTabIsInView] = useState(true);
    const [firstTabIsInView, setFirstTabIsInView] = useState(true);
    const tabInViewRefs = useRef<boolean[]>(new Array(Children.count(children)).fill(false));

    const { breakpointIndex } = useResponsiveValue();
    const isLaptop = breakpointIndex > 1;

    const setTabRef = (id: string) => (node: HTMLElement | null) => {
      if (!node) {
        return;
      }

      tabRefs.current[id] = node;
    };

    useEffect(() => {
      if (!tabRefs.current[tabId] || !tabGroupRef.current) {
        return;
      }

      const defaultOffset =
        ((tabGroupRef.current.children[0] as HTMLElement)?.offsetLeft ?? 0) - tabGroupRef.current.offsetLeft;

      tabGroupRef.current.scrollTo({
        left:
          tabRefs.current[tabId].offsetLeft -
          defaultOffset -
          tabGroupRef.current.offsetWidth / 2 +
          tabRefs.current[tabId].offsetWidth / 2,
        behavior: 'smooth',
      });
    }, [tabId]);

    useEffect(() => {
      tabInViewRefs.current = new Array(Children.count(children)).fill(false);
    }, [children]);

    const goToPrevPage = () => {
      const firstTabInViewIndex = tabInViewRefs.current.indexOf(true);
      const targetTabIndex = Math.max(firstTabInViewIndex - 1, 0);
      const targetTabNode = tabRefs.current[tabElements[targetTabIndex].props.id];

      if (!targetTabNode) {
        return;
      }

      tabGroupRef.current?.scrollTo({
        left: targetTabNode.offsetLeft - (tabGroupRef.current.offsetWidth - targetTabNode.offsetWidth) + 40,
        behavior: 'smooth',
      });
    };

    const goToNextPage = () => {
      const lastTabInViewIndex = tabInViewRefs.current.lastIndexOf(true);
      const targetTabIndex = Math.min(lastTabInViewIndex + 1, tabInViewRefs.current.length - 1);
      const targetTabNode = tabRefs.current[tabElements[targetTabIndex].props.id];

      if (!targetTabNode) {
        return;
      }

      tabGroupRef.current?.scrollTo({
        left: targetTabNode.offsetLeft - 40,
        behavior: 'smooth',
      });
    };

    const isScrollingRef = useRef(false);
    // eslint-disable-next-line @typescript-eslint/naming-convention
    const clientXRef = useRef<number | null>(null);
    const scrollLeftRef = useRef<number | null>(null);

    const startScroll: MouseEventHandler = e => {
      isScrollingRef.current = true;

      clientXRef.current = e.clientX;

      scrollLeftRef.current = tabGroupRef.current ? tabGroupRef.current.scrollLeft : null;
    };

    const scrollOnMove: MouseEventHandler = e => {
      if (!isScrollingRef.current || clientXRef.current === null || scrollLeftRef.current === null) {
        return;
      }

      tabGroupRef.current?.scrollTo({ left: scrollLeftRef.current - (e.clientX - clientXRef.current) });
    };

    const resetScroll: MouseEventHandler = () => {
      isScrollingRef.current = false;
    };

    return (
      <VStack width="100%">
        <BoxComponent
          as="ul"
          ref={tabGroupRef}
          onMouseDown={startScroll}
          onMouseMove={scrollOnMove}
          onMouseUp={resetScroll}
          onMouseLeave={resetScroll}
          flexDirection="row"
          mb={-1}
          {...restProps}
        >
          <Space width={[20, 20, 0]} />
          {tabElements.map((element, index) =>
            isLaptop ? (
              <InView
                key={element.props.id}
                onChange={inView => {
                  if (index === 0) {
                    setFirstTabIsInView(inView);
                  }

                  if (index === tabElements.length - 1) {
                    setLastTabIsInView(inView);
                  }

                  tabInViewRefs.current[index] = inView;
                }}
                options={
                  index === 0 || index === tabElements.length - 1
                    ? { threshold: 0.99, root: tabGroupRef.current }
                    : { threshold: 0.99, rootMargin: '0px -40px 0px -40px', root: tabGroupRef.current }
                }
              >
                <HStack
                  as="li"
                  mr={index !== tabElements.length - 1 ? [20, 20, 28] : 0}
                  flexGrow={tabFlexGrow}
                  flexShrink={tabFlexShrink}
                  flexBasis={tabFlexBasis}
                  hidden={element.props.hidden}
                  ref={setTabRef(element.props.id)}
                >
                  {cloneElement(element, {
                    active: element.props.id === tabId,
                    onClick: onTabChange,
                  })}
                </HStack>
              </InView>
            ) : (
              <HStack
                as="li"
                key={element.props.id}
                mr={index !== tabElements.length - 1 ? [20, 20, 28] : 0}
                flexGrow={tabFlexGrow}
                flexShrink={tabFlexShrink}
                flexBasis={tabFlexBasis}
                hidden={element.props.hidden}
                ref={setTabRef(element.props.id)}
              >
                {cloneElement(element, {
                  active: element.props.id === tabId,
                  onClick: onTabChange,
                })}
              </HStack>
            )
          )}
          <Space width={[20, 20, 0]} />
        </BoxComponent>
        {isLaptop && !firstTabIsInView && (
          <HStack position="absolute" height="100%" left={0}>
            <Pressable
              ariaLabel="Go to previous page"
              onClick={goToPrevPage}
              width={40}
              height="100%"
              backgroundColor="background"
              px={12}
              justifyContent="center"
            >
              <Icon.ChevronLeft.Regular size={16} />
            </Pressable>
            <Paper width={6} height="100%" gradient="linearLeft" />
          </HStack>
        )}
        {isLaptop && !lastTabIsInView && (
          <HStack position="absolute" height="100%" right={0}>
            <Paper width={6} height="100%" gradient="linearRight" />
            <Pressable
              ariaLabel="Go to next page"
              onClick={goToNextPage}
              width={40}
              height="100%"
              backgroundColor="background"
              px={12}
              justifyContent="center"
            >
              <Icon.ChevronRight.Regular size={16} />
            </Pressable>
          </HStack>
        )}
        <Divider direction="horizontal" />
      </VStack>
    );
  }
);
