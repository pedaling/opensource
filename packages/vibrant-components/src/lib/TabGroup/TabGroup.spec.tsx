import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { baseTheme } from '@vibrant-ui/theme';
import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { Tab } from '../Tab';
import { TabGroup } from './TabGroup';

describe('<TabGroup />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;

  let mockScrollTo: () => void;

  beforeEach(() => {
    mockScrollTo = jest.fn();

    Element.prototype.scrollTo = mockScrollTo;
  });

  describe('when TabGroup rendered', () => {
    const mockTabChange = jest.fn(() => {});

    beforeEach(() => {
      renderer = render(
        <TabGroup data-testid="tabGroup" tabId="tab1" type="fullWidth" onTabChange={mockTabChange}>
          <Tab data-testid="tab1" id="tab1" title="tab1" />
          <Tab data-testid="tab2" id="tab2" title="tab2" />
          <Tab data-testid="hidden-tab" id="hiddenTab" title="hiddenTab" hidden={true} />
          <Tab data-testid="tab3" id="tab3" title="tab3" />
        </TabGroup>
      );

      element = renderer.getByTestId('tabGroup');
    });

    it('tab1 border bottom rendered', () => {
      const tab1 = renderer.getByTestId('tab1');

      expect(tab1).toHaveStyleRule('border-bottom-color', baseTheme.colors.light.outlineNeutral);
    });

    it('scrollIntoView called', () => {
      expect(mockScrollTo).toBeCalled();
    });

    describe('when tab2 clicked', () => {
      beforeEach(async () => {
        const tab2 = renderer.getByTestId('tab2');

        await waitFor(() => userEvent.click(tab2));
      });

      it('onTabChange with tab2 called', () => {
        expect(mockTabChange).toBeCalledWith('tab2');
      });
    });

    it('hidden tab spacing is hidden', () => {
      expect(element.children[3]).toHaveStyleRule('display', 'none');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
