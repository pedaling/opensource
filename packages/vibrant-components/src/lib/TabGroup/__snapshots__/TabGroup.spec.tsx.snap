// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabGroup /> when TabGroup rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  margin-bottom: -1px;
  overflow: hidden;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  width: 20px;
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    width: 0;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  margin-right: 20px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

@media screen and (min-width: 1024px) {
  .emotion-3 {
    margin-right: 28px;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  border-width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-bottom-color: #0c0c0c;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 10px;
  background-color: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
}

@media screen and (min-width: 1024px) {
  .emotion-5 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  border-width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-bottom-color: rgba(0, 0, 0, 0);
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 10px;
  background-color: inherit;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.25rem;
}

@media screen and (min-width: 1024px) {
  .emotion-8 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

.emotion-9 {
  display: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  margin-right: 20px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

@media screen and (min-width: 1024px) {
  .emotion-9 {
    margin-right: 28px;
  }
}

.emotion-10 {
  display: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  border-width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-bottom-color: rgba(0, 0, 0, 0);
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  padding-bottom: 10px;
  background-color: inherit;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
  position: relative;
  margin-right: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  border-width: 0;
  border-color: #00000026;
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 0;
  margin-bottom: 0;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="tabGroup"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <button
          class="emotion-4"
          data-testid="tab1"
        >
          <span
            class="emotion-5"
          >
            tab1
          </span>
        </button>
      </div>
      <div
        class="emotion-3"
      >
        <button
          class="emotion-7"
          data-testid="tab2"
        >
          <span
            class="emotion-8"
          >
            tab2
          </span>
        </button>
      </div>
      <div
        class="emotion-9"
      >
        <button
          class="emotion-10"
          data-testid="hidden-tab"
        >
          <span
            class="emotion-8"
          >
            hiddenTab
          </span>
        </button>
      </div>
      <div
        class="emotion-12"
      >
        <button
          class="emotion-7"
          data-testid="tab3"
        >
          <span
            class="emotion-8"
          >
            tab3
          </span>
        </button>
      </div>
      <div
        class="emotion-2"
      />
    </div>
    <div
      class="emotion-16"
    />
  </div>
</div>
`;
